lucene/test-framework/src/java/org/apache/lucene/util/LuceneTestCase.java

```diff
@@ -1414,6 +1415,9 @@ private static BaseDirectoryWrapper newFSDirectory(Path f, LockFactory lf, boole
       }
 
       Directory fsdir = newFSDirectoryImpl(clazz, f, lf);
+      if (rarely()) {
+        fsdir = new FileSwitchDirectory(Set.of("fdt", "fdx", "tim"), fsdir, newFSDirectoryImpl(clazz, f, lf), true);
```

Comment:
We could maybe randomize which extensions we add to the set here for better testing?

Reply:
I pushed a new commit